lib/chess/board.ex

```diff
@@ -30,22 +30,53 @@ defmodule Chess.Board do
     board["#{file},#{rank}"]
   end
 
-  def move_piece(board, %{"from" => from, "to" => to}) do
-    [from_file, from_rank] = from
-    [to_file, to_rank] = to
-
+  def move_piece(board, %{
+    "from" => [from_file, from_rank],
+    "to" => [to_file, to_rank]}
+  ) do
```

Comment:
Totally a thing of style, but I prefer the previous formatting in this. I tend to dislike the `def` spanning multiple lines because of pattern matching, especially if the pattern matching can be done inside the function without detracting from the meaning of the function.